library/Icinga/Crypt/RSA.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/* Icinga Web 2 | (c) 2020 Icinga Development Team | GPLv2+ */
+
+namespace Icinga\Crypt;
+
+use InvalidArgumentException;
+use UnexpectedValueException;
+
+/**
+ * Class RSA
+ *
+ * This class allows to encrypt and decrypt data.
+ *
+ * Usage
+ * -----
+ * <code>
+ * ** Encryption **
+ *
+ * $rsa = new RSA();
+ * $rsa->loadKey(...RSA::keygen());
+ *
+ * just Encrypt data
+ * $data = $rsa->encrypt($myVar1, myVar2); // multiple parameters allowed
+ *
+ * Encrypt and encodes the given data with MIME base64
+ * $data = $rsa->encryptToBase64($myVar1, myVar2); // multiple parameters allowed
+ *
+ *
+ * ** Decryption **
+ *
+ * $rsa = new RSA();
+ * $rsa->loadKey($publicKey, load_private_key_from_database($publicKey));
+ *
+ * Decrypt and decodes the given data which is encoded in MIME base64
+ * list($myVar1, $myVar2) = $rsa->decryptFromBase64(...$data);
+ * </code>
+ *
+ * @package Icinga\Crypt
+ */
+class RSA
+{
+    /** @var string */
+    private $pubKey;
+
+    /** @var string */
+    private $privKey;
+
+    /**
+     * Generates a new private and public key pair
+     *
+     * @return array
+     */
+    public static function keygen()
+    {
+        $res = openssl_pkey_new(
+            [
+                'digest_alg' => 'sha512',
+                'private_key_bits' => 4096,
+                'private_key_type' => OPENSSL_KEYTYPE_RSA,
+            ]
+        );
+        // Extract the private key from $res to $privKey
+        openssl_pkey_export($res, $privKey);
+        // Extract the public key from $res to $pubKey
+        $pubKey = openssl_pkey_get_details($res);
+        $pubKey = $pubKey["key"];
+
+        return [$privKey, $pubKey];
+    }
+
+    /**
+     * Get the public key
+     *
+     * @return string
+     */
+    public function getPublicKey()
+    {
+        if (empty($this->pubKey)) {
+            throw new UnexpectedValueException('No public key set');
+        }
+        return $this->pubKey;
+    }
+
+    /**
+     * Get the private key
+     *
+     * @return string
+     */
+    public function getPrivateKey()
+    {
+        if (empty($this->privKey)) {
+            throw new UnexpectedValueException('No private key set');
+        }
+        return $this->privKey;
+    }
+
+    /**
+     * Decrypt the given data
+     *
+     * @param mixed ...$data
+     *
+     * @return array
+     */
+    public function decrypt(...$data)
+    {
+        if (empty(array_filter($data))) {
```

Comment:
This counts for all encryption/decryption functions:
Please don't filter the input data. If the function is called without parameters, we just return the empty array.
With `array_filter` you deny values which should be accepted by our functions.
Please add test cases for the following:

* The encrypt/decrypt methods return the empty array if no parameters are given
* Parameters that are considered empty by `empty()` or considered `false` in `array_filter` must work

Please also check the decrypted value of an encrypted bool `false`.